flake.nix

```diff
@@ -45,10 +46,13 @@
           rust
           rust-analyzer-unwrapped
           pkg-config
-          xorg.libX11
           gtk4
           libadwaita
+        ] ++ lib.optionals stdenv.isLinux [
+          xorg.libX11
```

Comment:
I would like to have this in a separate PR, as it is not directly related to this one